pkg/frontend/frontend_series.go

```diff
@@ -2,15 +2,46 @@ package frontend
 
 import (
 	"context"
+	"time"
 
 	"github.com/bufbuild/connect-go"
+	"github.com/grafana/dskit/tenant"
+	"github.com/prometheus/common/model"
 
 	querierv1 "github.com/grafana/pyroscope/api/gen/proto/go/querier/v1"
 	"github.com/grafana/pyroscope/api/gen/proto/go/querier/v1/querierv1connect"
 	"github.com/grafana/pyroscope/pkg/util/connectgrpc"
+	"github.com/grafana/pyroscope/pkg/validation"
 )
 
 func (f *Frontend) Series(ctx context.Context, c *connect.Request[querierv1.SeriesRequest]) (*connect.Response[querierv1.SeriesResponse], error) {
 	ctx = connectgrpc.WithProcedure(ctx, querierv1connect.QuerierServiceSeriesProcedure)
+
+	tenantIDs, err := tenant.TenantIDs(ctx)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	}
+
+	if c.Msg.Start == 0 || c.Msg.End == 0 {
+		const defaultInterval = 4 * time.Hour
+		now := model.Now()
+		c.Msg.Start = now.Add(-defaultInterval).Unix()
+		c.Msg.End = now.Unix()
```

Comment:
I would prefer to not do that, ideally we should maintain the legacy behaviour of only  returning `head.Series` result when no Start/End supplied.

Reply:
I reverted this in https://github.com/grafana/pyroscope/pull/2431/commits/f508ad37d627a4ccecd1193786544dcad2ee5907